src/AWS.Deploy.Orchestration/DeploymentBundleHandler.cs

```diff
@@ -157,13 +166,21 @@ private string GetDockerExecutionDirectory(Recommendation recommendation)
             return dockerExecutionDirectory;
         }
 
-        private string GetDockerFilePath(Recommendation recommendation)
+        public string GetDockerfilePath(Recommendation recommendation)
         {
-            var dockerFileDirectory = new FileInfo(recommendation.ProjectPath).Directory?.FullName;
-            if (dockerFileDirectory == null)
-                throw new InvalidProjectPathException(DeployToolErrorCode.ProjectPathNotFound, "The project path is invalid.");
+            // User specified a path to a dockerfile 
+            if (!string.IsNullOrEmpty(recommendation.DeploymentBundle.DockerfilePath))
```

Comment:
This is now calling ` DockerUtilities.TryGetDockerfile`, which does check if the file still exists.

Though, currently it'll still pass an empty string into `docker build` which would then fail. I wasn't sure if we should add another exception at this point, since either way the build isn't going to be successful.

Reply:
It is a good idea to add an exception is the file doesn't exist as we know for sure the `docker build` would fail. If we leave it till that point, the exception we get will be a generic docker build exception which will not track well in telemetry.